File: Esercitazione_4/2/conn_reuse_vers/lpd-vers/Prova.java
public class Prova {
    public static void main(String[] args) {
        int string_len;
        byte[] len = new byte[2];
        String string = "HELLO GUYS";

        /** HELLO GUYS = 10 caratteri/bytes */
        string_len = string.length();

        /** Codifica della lunghezza in Network Byte Order */
        len[0] = (byte)((string_len & 0xFF00) >> 8);
        len[1] = (byte)(string_len & 0x00FF);

        /** Decodifica della lunghezza */
        int len_fin = (len[0] << 8) | len[1];

        System.out.println("La lunghezza della stringa è: " + len_fin + " caratteri");
    }
}

File: Esercitazione_4/2/conn_reuse_vers/lpd-vers/prova.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char const *argv[])
{
	int string_len;
	int len[2];
	char string[1024];

	/* HELLO GUYS = 10 caratteri/bytes */
	strcpy(string, "HELLO GUYS");

	string_len = strlen(string);
	/* Codifica della lunghezza */
	len[0] = (string_len & 0xFF00) >> 8;
	len[1] = (string_len & 0x00FF);

	/* Decodifica della lunghezza */
	int len_fin = len[0] << 8 | len[1];

	printf("La lunghezza della stringa è: %d\n", len_fin);

	return 0;
}
